tests/components/prometheus/test_init.py

```diff
@@ -87,6 +88,64 @@
 PROMETHEUS_PATH = "homeassistant.components.prometheus"
 
 
+class EntityMetric:
+    """Represents a Prometheus metric for a Home Assistant entity."""
+
+    metric_name: str
+    labels: dict[str, str]
+
+    def __init__(self, metric_name: str, **kwargs: Any) -> None:
+        """Create a new EntityMetric based on metric name and labels."""
+        self.metric_name = metric_name
+        self.labels = kwargs
+
+        # Labels that are required for all entities.
+        for labelname in ("domain", "friendly_name", "entity"):
+            assert labelname in self.labels
+            assert self.labels[labelname] != ""
+
+    def withValue(self, value: float):
+        """Return a metric with value."""
+        return EntityMetricWithValue(self, value)
+
+    @property
+    def _metric_name_string(self) -> str:
+        """Return a full metric name as a string."""
+        labels = ",".join(
+            f'{key}="{value}"' for key, value in sorted(self.labels.items())
+        )
+        return f"{self.metric_name}{{{labels}}}"
+
+    def assertInBody(self, body: list[str]):
+        """Assert that this metric exists in the provided Prometheus output."""
+        assert any(line.startswith(self._metric_name_string) for line in body)
+
+    def assertNotInBody(self, body: list[str]):
+        """Assert that this metric does not exist in Prometheus output."""
+        assert self._metric_name_string not in body
```

Comment:
Note that as written assertNotInBody will never match, because Prometheus output always includes values.

Perhaps add an internal method for the actual check (`_in_metrics`), and then use that from both assert functions? This will ensure that the logic always matches, and will allow `EntityMetricWithValue` to only override `_in_metrics`, using assert methods from the parent class.

Also a few cosmetic suggestions (sorry for not thinking about these earlier):
* it's not very intuitive what `body` refers to, since this is so far away from the code that passes HTTP response body to these methods. Let's call it `metrics` instead? Or `metric_response` if you want to be more explicit?
* camelCase is not very idiomatic in Python, I suspect it's better use underscores (surprised that there is no linter checking this?)

```suggestion
    def _in_metrics(self, metrics: list[str]) -> bool:
        """Report whether this metric exists in the provided Prometheus output."""
        return any(line.startswith(self._metric_name_string) for line in body)

    def assert_in_metrics(self, metrics: list[str]):
        """Assert that this metric exists in the provided Prometheus output."""
        assert self._in_metrics(metrics)

    def assert_not_in_metrics(self, metrics: list[str]):
        """Assert that this metric does not exist in Prometheus output."""
        assert not self._in_metrics(metrics)
```